encode return value as integer instead of string

shibumi opened the pull request:
The in-toto python reference implementation encodes the return-value
field of our link files as integer. Our specification specified this
field as string value. This commit fixes this inconsistency and
assigns the right data type to the return-value field.

CC: @SantiagoTorres @lukpueh @trishankatdatadog

shibumi commented:
One open question is, how do we want to handle null values? Right now I have just assigned `null` to the example of the link file, underlining that return-value is an integer object and not a string (the value `" "` would be wrong in this case). Practically, I think we never have no return value, right?

lukpueh commented on in-toto-spec.md:
```diff
@@ -1685,11 +1686,11 @@ link metadata:
        "vcs.log": { "sha256": "e64589ab156f325a4ab2bc5d532737d5a7..."}
     },
-    "return-value": "0",
+    "return-value": 0,
```
This seems like a remainder from when `return-value` was a 1st level field of `signed`. I suggest to remove this.

shibumi replied:
ah good catch! I didn't even realize that there is a return-value outside of the actual byproducts object.

lukpueh commented on in-toto-spec.md:
```diff
@@ -935,7 +935,8 @@ information about the step performed. Byproducts are not verified by in-toto’s
 dictionary should have standard output (stdout), standard error (stderr) and
-return value (return-value), even if no values are filled in.
+return value (return-value), even if no values are filled in. The return value
+is stored as integer value.
```
Maybe we can make this a *should be stored*, given that we claim `byproducts` is an opaque dictionary.

shibumi commented:
@SantiagoTorres @lukpueh I have force pushed a new version + I have modified the commit message, that it's clear we have also removed one return-value line that was a remainder.

lukpueh commented:
> Maybe @SantiagoTorres wants to make the final call. :)

re-ping @SantiagoTorres

adityasaky reviewed (commented):
I think this change looks good to me, and would represent the present behaviour in in-toto-python and in-toto-golang. Would we need to make any changes to record return-value as null if it isn't recorded?

shibumi commented:
@lukpueh  still need to withdraw his request for changes or click on 'approve', if this should get merged :)

lukpueh commented:
> Would we need to make any changes to record return-value as null if it isn't recorded?

Good question. in-toto-run always records a return-value in both implementations. in-toto-record (only available in the reference implementation) returns an empty byproducts field. Given that this is only a recommendation, I think we are fine either way.

lukpueh reviewed (approved):
Thanks for taking care of this, @shibumi!

adityasaky reviewed (approved):
Agreed, @lukpueh. LGTM!
